Dockerfile

```diff
@@ -0,0 +1,123 @@
+FROM ubuntu:jammy
+
+ENV DEBIAN_FRONTEND noninteractive
+ARG BRANCH_NAME
+
+RUN \
+  apt-get update && \
+  apt-get -y upgrade
+
+RUN \
+  apt-get -y --no-install-recommends install \
+    build-essential \
+    ca-certificates \
+    cmake \
+    curl \
+    git \
+    libc6-dbg \
+    libgtest-dev \
+    linux-libc-dev \
+    ninja-build \
+    pkg-config \
+    python3 \
+    tar \
+    unzip \
+    vim \
+    wget \
+    zip
+
+# Download the Android NDK.
+RUN \
+  wget https://dl.google.com/android/repository/android-ndk-r25c-linux.zip && \
+  unzip android-ndk-r25c-linux.zip && \
+  rm android-ndk-r25c-linux.zip
+ENV ANDROID_NDK_HOME /android-ndk-r25c
+
+# This is needed by OpenSSL because for some reason it uses PATH to find the Android NDK compilers (and obviously
+# this will only work for linux-x86_64 images).
+ENV PATH $ANDROID_NDK_HOME/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH
+
+# Clone the repository.
+RUN git clone https://github.com/hashgraph/hedera-sdk-cpp.git
+
+# Change the working directory to that of the cloned repository.
+WORKDIR /hedera-sdk-cpp
+
+# Checkout the working branch.
+RUN git checkout origin/$BRANCH_NAME
+
+# Update the submodule.
+RUN git submodule update --init
+
+# Build
+RUN cmake --preset android-arm64-release
+RUN cmake --preset android-arm64-debug
+RUN cmake --build --preset android-arm64-release
+RUN cmake --build --preset android-arm64-debug
+
+# TODO:
+# The steps below are steps needed to test the Android library executables. Currently, the Android emulator is unable to
+# run in a linux-x86_64 environment because it is built for arm64. If there is interest in testing the Android library
+# in CI, this issue of where to run the Android emulator will need to be figured out (must be in an arm64 environment,
+# arm64 docker container has been tested but doesn't work). The steps here should be the general workflow of how to get
+# the emulator up and running and the test executables ran.
+
+# Download Android SDK
```

Comment:
Maybe these can go into a makefile so we don't leave them just commented out in this docker file.

Reply:
When/if these are implemented, the commands are going to have to be run in the Dockerfile anyways so I don't really see any sense in moving them, unless I'm misunderstanding?